### ISSUE TITLE
Support for SegmentTemplate URLs?

### ISSUE THREAD

**Issue by JoeyMorani** [sergey-dryabzhinsky/nginx-rtmp-module#140]
Hi,

Are there any plans to implement SegmentTemplate URLs in addition to SegmentTimeline to fully and properly support seeking with a large `timeShiftBufferDepth` using a dynamic/live manifest? SegmentTimeline isn't suitable (from what I've seen and been advised here - https://github.com/Dash-Industry-Forum/dash.js/issues/1405) due to the listing of every segment in the manifest and the constant refreshes. 

I've modified ngx_rtmp_dash_module.c to generate a SegmentTemplate MPD however it appears that segments aren't always created in sync with the `dash_fragment` duration which causes live playback issues/404 errors. Here's an example SegmentTemplate MPD which is generated:

```
<?xml version="1.0"?>
<MPD
    type="dynamic"
    xmlns="urn:mpeg:dash:schema:mpd:2011"
    availabilityStartTime="2016-05-28T18:33:59Z"
    minimumUpdatePeriod="P100Y"
    minBufferTime="PT1S"
    timeShiftBufferDepth="PT24H"
    profiles="urn:mpeg:dash:profile:isoff-live:2011,http://dashif.org/guidelines/dash-if-simple"
    xsi:schemaLocation="urn:mpeg:DASH:schema:MPD:2011 DASH-MPD.xsd"
    xmlns:xsi="http://www.w3.org/2011/XMLSchema-instance">
  <Period start="PT0S" id="dash">
    <AdaptationSet
        id="1"
        contentType="video"
        segmentAlignment="true"
        maxWidth="1920"
        maxHeight="1080"
        mimeType="video/mp4"
        maxFrameRate="62500/1000"
        startWithSAP="1">
      <Representation
          id="test_H264"
          codecs="avc1.64002a"
          width="1920"
          height="1080"
          frameRate="62500/1000"
          sar="1:1"
          bandwidth="3000000"></Representation>
        <SegmentTemplate
            timescale="1000"
            duration="1000"
            initialization="init.m4v"
            media="$Number$.m4v"
            startNumber="0">
        </SegmentTemplate>
    </AdaptationSet>
    <AdaptationSet
        id="2"
        contentType="audio"
        mimeType="audio/mp4"
        segmentAlignment="true"
        startWithSAP="1">
      <Representation
          id="test_AAC"
          codecs="mp4a.40.2"
          audioSamplingRate="44100"
          bandwidth="256000">
          <AudioChannelConfiguration
              schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011"
              value="2" />
      </Representation>
        <SegmentTemplate
            timescale="1000"
            duration="1000"
            initialization="init.m4a"
            media="$Number$.m4a"
            startNumber="0">
        </SegmentTemplate>
    </AdaptationSet>
  </Period>
</MPD>
```

Basically, I wish to provide live dash video/audio with the ability to seek backwards over ALL previous segments using dash.js and this module. If anyone has any advice on how to resolve this problem it would very much be appreciated.

**Comment by JoeyMorani** [comment]
Hey,

Any feedback on this? I've still not had any luck.

I want dash fragments generated by nginx-rtmp to be no longer than the specified `dash_fragment` duration to ensure that they aren't loaded by dash.js when they don't yet exist. I've attempted to use ffmpeg to force keyframes at the correct interval before passing it to nginx-rtmp but this doesn't work.

Any ideas would be appreciated.
